src/core/sys/posix/pthread.d

```diff
@@ -499,8 +499,11 @@ int pthread_key_delete(pthread_key_t);
 int pthread_mutex_destroy(pthread_mutex_t*);
 int pthread_mutex_init(pthread_mutex_t*, pthread_mutexattr_t*) @trusted;
 int pthread_mutex_lock(pthread_mutex_t*);
+int pthread_mutex_lock(shared(pthread_mutex_t)*);
```

Comment:
I deliberately made that change because it bothered me when porting C and C++ code to D, which was directly using the platform specific APIs. Especially when porting a C project where there were absolutely zero abstractions and I needed to add casts to almost every function, since almost every function used `pthread_mutex` directly and I wanted to use `shared` instead of `__gshared`.

> I'm surprised that this even works

You shouldn't be, if you consider that [`strchr`](http://en.cppreference.com/w/cpp/string/byte/strchr) works the same way in C++. In C all symbols (including functions) are mangled the same as their name (give or take an underscore at the beginning). That allows (sometimes correctly and sometimes not) to map multiple overloads from C++ or D to the same C function. (Note: some C++ compilers allow C function overloading but require vendor specific extensions similar to D's `pragma (mangled)`.)

Also consider that in D you can overload `extern (C)` functions on attributes like nothrow. Especially useful for function pointer parameters.